sogs/routes/rooms.py

```diff
@@ -960,3 +960,27 @@ def delete_all_posts(room, sid):
     if not deleted:
         abort(http.NOT_FOUND)
     return jsonify({})
+
+@rooms.delete("/room/all/<SessionID:sid>")
+def delete_user_posts_from_all_rooms(sid):
+    """
+    Deletes all posts made by a user from all rooms
+
+    # URL Parameters
+
+    - `sid` — the session id of the user to ban
+
+    # Return value
+
+    An empty json object is returned.
+
+    # Error status codes
+
+    - 403 Forbidden — if the invoking user does not have access to a room.
+    """
+    user = muser.User(session_id=sid, autovivify=False)
+    for room in mroom.get_accessible_rooms(g.user):
+        deleted, _ = room.delete_all_posts(user, deleter=g.user)
+        if not deleted:
+            app.logger.warning(f"Error deleting posts from room: {room}.")
+    return jsonify({})
```

Comment:
It would be nice to return a summary of deletion counts here so that a client could display the summary; perhaps something like:
```Python
deletions = {}
total = 0
for room in ...
    try:
        count, _ = room.delete_all_posts(...
        total += count
        deletions[room.token] = count
    # ...

return jsonify({"total": total, "rooms": deletions})

Reply:
A great example of how to take a simple snippet of code and make it so much better.

Thank you for all these suggestions.

Changed as described.